server/models/schema.sql

```diff
@@ -98,6 +104,7 @@ CREATE TABLE `cash` (
   `description`     TEXT,
   `is_caution`      BOOLEAN NOT NULL DEFAULT 0,
   PRIMARY KEY (`uuid`),
+  UNIQUE KEY `cash_1` (`reference`, `project_id`),
```

Comment:
You are right, it is ensure by the triggers but it is not a rule so, it is possible for an other extra component as (Symmetric) to insert a duplication. We are loosing nothing by including this rule, we are safe instead.